### PR TITLE
Abstract out api.pod spec

### PR DESCRIPTION
**Pull request by procrypt**
Fixes: #348 
cc: @containscafeine @kadel @cdrage @surajssd

### CODE REVIEW

**Comment by kadel** [comment]
PodSpec is also used in OpenShift transformer. It would be great to use podspec function also there.

File: pkg/transformer/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -104,6 +104,19 @@ func (k *Kubernetes) CheckUnsupportedKey(komposeObject *kobject.KomposeObject, u
 	return keysFound
 }
 
+// podspec creates the pod specification
+func (k *Kubernetes) podspec(name string, service kobject.ServiceConfig) api.PodSpec {
```
Comment (cdrage) [inline pkg/transformer/kubernetes/kubernetes.go]: should be podSpec (Go name conventions)

Comment (kadel) [inline pkg/transformer/kubernetes/kubernetes.go]: and even better might be initPodSpec similar to other init functions ;-)

**Comment by procrypt** [comment]
@kadel We can't use `PodSpec` function in OpenShift because `PodSpec` in OpenShift is different.
PodSpec in `Kubernetes`
```go
	pod := api.PodSpec{
		Containers: []api.Container{
			{
				Name:  name,
				Image: service.Image,
			},
		},
	}
```

PodSpec in `OpenShift`
```go
	pod := api.PodSpec{
		Containers: []api.Container{
			{
				Name:  name,
				Image: " ",
			},
		},
	}
```

**Comment by kadel** [comment]
I think that if we are abstracting podSpec in one place it would be great to have everywhere else.
 
If you change your `initPodSpec` function it can be used also in OpenShift. 
You don't have to pass whole `service kobject.ServiceConfig` struct down to `initPodSpec`.
All you need is `name` and `image`. It could look like this:

```go
func (k *Kubernetes) InitPodSpec(name string, image string) api.PodSpec {
... 
```

**Comment by procrypt** [comment]
@kadel PR updated.

File: pkg/transformer/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -104,6 +104,19 @@ func (k *Kubernetes) CheckUnsupportedKey(komposeObject *kobject.KomposeObject, u
 	return keysFound
 }
 
+// "InitPodSpec creates the pod specification"
```
Comment (kadel) [inline pkg/transformer/kubernetes/kubernetes.go]: remove `"` and it will be ok ;-)

Comment (procrypt) [inline pkg/transformer/kubernetes/kubernetes.go]: @kadel tests are passing locally when I run `make test`, no idea what is going wrong in Travis.

**Comment by cdrage** [comment]
LGTM

**Pull request merged**